src/env.h

```diff
@@ -54,6 +54,12 @@ class performance_state;
 
 namespace loader {
 class ModuleWrap;
+struct PackageConfig {
+  const bool exists;
+  const bool is_valid;
+  const bool has_main;
+  const std::string main;
+};
```

Comment:
For legibility reasons, it might be better to use enums instead of bools:
```c++
struct PackageConfig {
  const enum { Exists, DoesNotExist } exists;
  const enum { Valid, Invalid } is_valid;
  const enum { HasMain, NoMain } has_main;
  const std::string main;
};
// init with PackageConfig { PackageConfig::Exists, PackageConfig::Valid, PackageConfig::NoMain, "" }
```
Adds a little extra type safety because the enums are distinct types, i.e., you can't assign `Invalid` to `.exists`.

There's an alternative version that looks like this:
```c++
struct Exists {
  enum Bool { Yes, No };
};

struct IsValid {
  enum Bool { Yes, No };
};

struct HasMain {
  enum Bool { Yes, No };
};

struct PackageConfig {
  const Exists::Bool exists;
  const IsValid::Bool is_valid;
  const HasMain::Bool has_main;
  const std::string main;
};
// init with PackageConfig { Exists::Yes, IsValid::Yes, HasMain::No, "" }
```
I personally prefer the second one but it's slightly more laborious to write.